### PR TITLE
Add a new mechanism for rendering rfc7951 JSON.

### PR DESCRIPTION
**Pull request by robshakir**
```
 * (M) ygot/{render,render_test}.go
  - Some users wish to be able to render a specific field to
    RFC7951 JSON directly, such that a particular subtree can be
    rendered.
    Today, this is not possible since we require arguments to the
    rendering functions to be specified as GoStructs. This PR adds
    a new rendering method that specifically takes an interface{}
    and returns a string such that these users can call this method
    generically for any valid input type of a ygen generated Go file.
    We lose some type-safety in this mechanism, but this is similar to
    the interface provided by json.Marshal.
    This method is implemented only for RFC7951 JSON, as we aim to
    move away from the 'internal' format (which today is only
    supported for marshalling and not unmarshalling).
```

### CODE REVIEW

**Comment by robshakir** [comment]
Big thumbs up for `staticcheck` here @wenovus -- it pointed something out that I'd missed, which meant that I could really simplify this code. Thanks for setting it up!

File: ygot/render_test.go [hunk 1]
```diff
@@ -2903,3 +2903,95 @@ func TestFindUpdatedLeaves(t *testing.T) {
...
+		in:   map[string]*renderExample{},
+		want: `null`,
+	}, {
+		desc: "nil string pointer",
+		in: func() *string {
+			var s *string
+			return s
+		}(),
+		want: `null`,
+	}}
```
Comment (wenovus) [inline ygot/render_test.go:2980]: Add test for empty type

```suggestion
        }, {
                desc: "empty type",
                in: &renderExample{
                        Empty: true,
                },
                want: `{"empty":[null]}`,
	}}
```

File: ygot/render.go [hunk 2]
```diff
@@ -884,6 +884,27 @@ func ConstructInternalJSON(s GoStruct) (map[string]interface{}, error) {
 	})
 }
 
+// EmitRFC7951 renders the supplied interface to RFC7951-compatible JSON. The argument
+// supplied must be a valid type within a generated ygot GoStruct - but can be a member
```
Comment (wenovus) [inline ygot/render.go:888]:
```suggestion
// supplied must be a valid type within a generated ygot GoStruct (i.e. all types except the root GoStruct type) - but can be a member
```

File: ygot/render_test.go [hunk 3]
```diff
@@ -2903,3 +2903,95 @@ func TestFindUpdatedLeaves(t *testing.T) {
...
+	}, {
+		desc: "empty map",
+		in:   map[string]*renderExample{},
+		want: `null`,
+	}, {
+		desc: "nil string pointer",
+		in: func() *string {
+			var s *string
+			return s
+		}(),
```
[the remark this answers is not in the document]
Comment (robshakir) [reply ygot/render_test.go:2978]: Thanks!

File: ygot/render_test.go (hunk 1 above)
Comment (robshakir) [reply ygot/render_test.go:2980]: Great, thanks! Added.

File: ygot/render.go (hunk 2 above)
Comment (robshakir) [reply ygot/render.go:888]: Can you explain what you meant here please? I'm not sure I understood what `root` means in this context?

**Comment by robshakir** [comment]
Thanks for the reviews.

@ejbrever - PTAL, I added arguments (trying to keep a consistent behaviour here despite the fact that we don't have the same arguments as `EmitJSON`). It now works similarly to other functions that take an expanded slice of arguments, so you can say:

```golang
ygot.Marshal7951(d, JSONIndent("\t"), &RFC7951Config{AppendModuleNames: true})
```

This also cleans up the base syntax such that you can just do:

```
ygot.Marshal7951(d)
```
 
without needing the extraneous `nil` at the end.

File: ygot/render.go (hunk 2 above)
Comment (wenovus) [reply ygot/render.go:906]: I just meant that this cannot be used for the root of all the models, because from my reading of the code it seems that should be handled by `EmitJSON`.

File: ygot/render.go [hunk 4]
```diff
@@ -884,6 +888,66 @@ func ConstructInternalJSON(s GoStruct) (map[string]interface{}, error) {
...
+// Marshal7951 renders the supplied interface to RFC7951-compatible JSON. The argument
+// supplied must be a valid type within a generated ygot GoStruct - but can be a member
+// field of a generated struct rather than the entire struct - allowing specific fields
+// to be rendered. The supplied arguments control the JSON marshalling behaviour - both
+// base JSON Marshal (e.g., indentation), as well as RFC7951 specific options such as
+// YANG module names being appended.
+// The rendered JSON is returned as a byte slice - in common with json.Marshal.
+func Marshal7951(d interface{}, args ...Marshal7951Arg) ([]byte, error) {
+	var (
+		rfccfg *RFC7951JSONConfig
```
Comment (ejbrever) [inline ygot/render.go:914]: big nit...can you make this var rfcCfg just to help readability a little.

Comment (robshakir) [reply ygot/render.go]: Of course -- done!
